kernel/device.c

```diff
@@ -143,3 +159,18 @@ const char * device_name( struct device *d )
 {
 	return d->driver->name;
 }
+
+void device_driver_get_stats(char * name, struct device_driver_stats * s)
```

Comment:
Please factor out the matter of finding a driver by name into a separate function:
```
struct device_driver * device_driver_lookup( const char *name )
```

And then also use that in `driver_open`.